### PR TITLE
Add a `keyspace` configuration in the `vschema` for foreign key mode

### LINKED ISSUE #13552: Add a keyspace configuration in the vschema for foreign keys
### Description

As we kick off the work on foreign keys, we will need a vschema configuration that tells us what foreign key mode the user desires.

### PR DESCRIPTION
**Pull request by GuptaManan100**
## Description

As we kick off the work on foreign keys, we will need a vschema configuration that tells us what foreign key mode the user desires. We have chosen to add an enum for this purpose that takes in the following 4 values (for now) - 
1. Default Value - This has been added so that we can change what the default value is down the road.
2. Disallow mode - Vitess will disallow creation of foreign keys. (This is intended to sunset the flag `foreign_key_mode`)
3. Unmanaged mode - Vitess will not interfere with any query and let MySQL and the user figure out the correctness and constraints.
4. Managed mode - Vitess will intervene in between to ensure that cascades are broken up so that vreplication continues to work.

## Related Issue(s)

- Fixes #13552 

## Checklist

-   [x] "Backport to:" labels have been added if this change should be back-ported
-   [x] Tests were added or are not required
-   [x] Did the new or modified tests pass consistently locally and on the CI
-   [x] Documentation was added or is not required

### CODE REVIEW

**Comment by GuptaManan100** [comment]
I have not added release notes changes or docs for this for 2 reasons - 
1. We have a separate item for that in https://github.com/vitessio/vitess/issues/12967
2. We aren't sure at this point if the entire work required to support foreign keys will complete in v18 or not. If not, we would not want to document these parameters and mislead the users.

File: proto/vschema.proto [hunk 1]
```diff
@@ -45,6 +45,15 @@ message Keyspace {
   map<string, Table> tables = 3;
   // If require_explicit_routing is true, vindexes and tables are not added to global routing
   bool require_explicit_routing = 4;
+  // foreign_key_mode dictates how Vitess should handle foreign keys for this keyspace.
+  ForeignKeyMode foreign_key_mode = 5;
+
+  enum ForeignKeyMode {
+    Fk_default = 0;
```
Comment (deepthi) [inline proto/vschema.proto:52]: Enum value identifiers are mostly uppercase with underscores, though people have introduced a few that use CamelCase. Can you change these to be similar to this example? We don't need a third way of naming these.
```
DEFAULT_PLANNER = 0;
```

Comment (deepthi) [inline proto/vschema.proto:52]: What's the idea of `default`? People will automatically get new behavior when we change the default behavior? I don't think this is such a good idea. Opting in and out should be explicit and new users who don't want to choose can get whatever the default is in that release.

Comment (GuptaManan100) [reply proto/vschema.proto]: I have changed the case to be all caps-lock 👍

**Review by frouioui — APPROVED** [review]

**Review by deepthi — APPROVED** [review]

**Pull request merged**